File: tools/build_phone_csv_from_json.js
#!/usr/bin/env node
// Build CSV (phone,memberId,displayName) from docs/cleaned_members_with_phones.json
// Usage: node tools/build_phone_csv_from_json.js [--input=./docs/cleaned_members_with_phones.json] [--output=./tools/members_from_json.csv]

const fs = require('fs');
const path = require('path');

function parseArgs() {
  const out = {};
  for (const arg of process.argv.slice(2)) {
    const m = arg.match(/^--([^=]+)=(.*)$/);
    if (m) out[m[1]] = m[2];
  }
  return out;
}

function normalizePhone(raw) {
  if (!raw) return '';
  let p = String(raw).trim();
  // Remove non-digits except leading +
  if (p.startsWith('+')) {
    p = '+' + p.slice(1).replace(/\D+/g, '');
  } else {
    p = p.replace(/\D+/g, '');
    if (p && !p.startsWith('+')) {
      // naive default to US +1 if 10 digits
      if (p.length === 10) p = '+1' + p;
      else if (!p.startsWith('+')) p = '+' + p; // best-effort
    }
  }
  return p;
}

async function main() {
  const args = parseArgs();
  const input = path.resolve(args.input || './docs/cleaned_members_with_phones.json');
  const output = path.resolve(args.output || './tools/members_from_json.csv');

  if (!fs.existsSync(input)) {
    console.error('Input JSON not found:', input);
    process.exit(1);
  }
  const raw = JSON.parse(fs.readFileSync(input, 'utf8'));
  if (!Array.isArray(raw)) {
    console.error('Input JSON must be an array.');
    process.exit(1);
  }
  const rows = [];
  rows.push('phone,memberId,displayName');
  for (const m of raw) {
    const memberId = m.memberId || m.id || '';
    const name = m.fullName || m.name || '';
    const phone = normalizePhone(m.phone || m.phoneNumber || '');
    if (!memberId || !phone) continue;
    rows.push([phone, memberId, JSON.stringify(String(name).replace(/\n/g, ' ').trim())].join(','));
  }
  fs.writeFileSync(output, rows.join('\n'));
  console.log('Wrote', output, 'rows:', rows.length - 1);
}

main().catch(e => { console.error(e); process.exit(1); });

File: aaac-membership-system/tools/build_phone_csv_from_json.py
#!/usr/bin/env python3
"""
Build CSV (phone,memberId,displayName) from docs/cleaned_members_with_phones.json

Usage:
  python3 tools/build_phone_csv_from_json.py \
    --input /absolute/path/to/docs/cleaned_members_with_phones.json \
    --output /absolute/path/to/tools/members_from_json.csv
"""

import argparse
import csv
import json
import os
import re

def normalize_phone(raw: str) -> str:
    if not raw:
        return ''
    p = str(raw).strip()
    if p.startswith('+'):
        p = '+' + re.sub(r'\D+', '', p[1:])
    else:
        p = re.sub(r'\D+', '', p)
        if p:
            if len(p) == 10:
                p = '+1' + p
            elif not p.startswith('+'):
                p = '+' + p
    return p

def main():
    ap = argparse.ArgumentParser()
    ap.add_argument('--input', required=True)
    ap.add_argument('--output', required=True)
    ap.add_argument('--set', action='append', default=[], help='Override phone in format memberId:phone (can repeat)')
    args = ap.parse_args()

    with open(args.input, 'r', encoding='utf-8') as f:
        data = json.load(f)
    if not isinstance(data, list):
        raise SystemExit('Input JSON must be an array')

    os.makedirs(os.path.dirname(args.output), exist_ok=True)
    # Build overrides map
    overrides = {}
    for item in args.set:
        try:
            mid_str, phone_raw = item.split(':', 1)
            mid_num = int(mid_str.strip())
            overrides[mid_num] = normalize_phone(phone_raw)
        except Exception:
            pass
    with open(args.output, 'w', newline='', encoding='utf-8') as f:
        w = csv.writer(f)
        w.writerow(['phone', 'memberId', 'displayName'])
        rows = 0
        for m in data:
            member_id = m.get('memberId') or m.get('id') or ''
            name = m.get('fullName') or m.get('name') or ''
            phone = normalize_phone(m.get('phone') or m.get('phoneNumber') or '')
            # filter: numeric memberId between 1..101 and phone present
            try:
                mid_num = int(str(member_id))
            except Exception:
                continue
            if mid_num < 1 or mid_num > 101:
                continue
            # apply override if provided
            if mid_num in overrides:
                phone = overrides[mid_num] or phone
            if not phone:
                continue
            w.writerow([phone, member_id, name.replace('\n', ' ').strip()])
            rows += 1
    print(f'Wrote {args.output} rows: {rows}')

if __name__ == '__main__':
    main()


